src/array_api_stubs/_draft/data_type_functions.py

```diff
@@ -40,8 +40,14 @@ def astype(
     ----------
     x: array
         array to cast.
-    dtype: dtype
-        desired data type.
+    dtype_or_kind: Union[dtype, str]
+        desired data type or kind of data type. Supported kinds are:
+        -   ``'bool'``: boolean data types (e.g., ``bool``).
+        -   ``'signed integer'``: signed integer data types (e.g., ``int8``, ``int16``, ``int32``, ``int64``).
+        -   ``'unsigned integer'``: unsigned integer data types (e.g., ``uint8``, ``uint16``, ``uint32``, ``uint64``).
+        -   ``'integral'``: integer data types. Shorthand for ``('signed integer', 'unsigned integer')``.
+        -   ``'real floating'``: real-valued floating-point data types (e.g., ``float32``, ``float64``).
+        -   ``'complex floating'``: complex floating-point data types (e.g., ``complex64``, ``complex128``).
```

Comment:
I included `integral` here, but left out `numeric`. Would be happy to have both or neither if that is preferred.

Reply:
I don't think `integral` can be included, since it's unclear whether to select signed or unsigned.